snooty/postprocess.py

```diff
@@ -527,18 +527,21 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             )
 
         if isinstance(node, n.Directive) and node.name == "collapsible":
-            html5_id = util.make_html5_id(node.options["heading"]).lower()
-            node.options["id"] = html5_id
+            self.current_depth += 1
+            if self.current_depth - 1 > self.contents_depth:
+                return
             self.headings.append(
                 ContentsHandler.HeadingData(
                     self.current_depth,
```

Comment:
Looks like we'll have a double depth increment here due to the section being added. This will mean that for an example like:

```
.. contents::
    :depth: 1

===================
Heading of the page
===================

.. collapsible::
    :heading: Collapsible heading
    :sub_heading: Subheading

    ~~~~~~~~~~~~~~~
    This is content
    ~~~~~~~~~~~~~~~
```

the user will have to bump `:depth:` to 3 instead of 2 if they want to capture `This is content` in the "On This Page" component.

We can probably just rely on the presence of the section to handle incrementing, while artificially adding 1 when adding the collapsible heading to account for the section being after the directive. What do you think?

```py
if isinstance(node, n.Directive) and node.name == "collapsible":
    self.headings.append(
        ContentsHandler.HeadingData(
            # Add 1 since section appears as a child
            self.current_depth + 1,
            node.options["id"],
            [n.Text(node.span, node.options["heading"])],
        )
    )
```

Reply:
genius! done! added above test case 👍